com.unity.netcode.gameobjects/Runtime/Messaging/MessageHeader.cs

```diff
@@ -0,0 +1,8 @@
+namespace Unity.Netcode
+{
+    public struct MessageHeader
```

Comment:
should this be public or can it be just an internal?

Reply:
also can we bake 1 byte + 1 short values into APIs or structs or is this really needed to be a self-contained struct?

Reply:
A lot of things I make public should be internal. I keep forgetting internal exists - please excuse me as C# isn't my home language and I'm still trying to get my habits correct for it.

Reply:
would recommend doing a pass on this ASAP so that it is clear when you're proposing a new API and when you're not. all public APIs need xml docs...
in particular I was looking at `Bind` and that attribute is extremely generically named if it's public, but now I don't know whether to give that type of feedback since maybe it was meant to be internal

Reply:
`Bind` is intended to be internal, yes. I'll make a pass on that after lunch.

Reply:
Renamed Bind to something more descriptive and made it internal. Can't make MessageHeader internal if INetworkMessage isn't internal - hoping we can provide INetworkMessage as our custom message implementation because I think it's a lot more usable than the existing one, but if we end up making that internal, I'll make this internal with it (as well as NetworkContext)